### PR TITLE
feat: update bazel documentation and enable `bazel-module` manager

### LINKED ISSUE #13658: Bazel: support bzlmod
### What would you like Renovate to be able to do?

Hi, @rarkins. Thanks for the wonderfully great tool that is Renovate. It's been spectacular for our Bazel repo--and we point other Bazel users to you guys every chance we get, including in Bazel tools we've open sourced.

I just stumbled upon some docs indicating that Bazel working to update their external dependency management to use a different, more unified file format, and has a working version out. They call the manager bzlmod. @alexeagle, who I know you've cc'd on Bazel issues before, wrote a great blog post on it. I'll just link, because I certainly can't do a better job describing it here. https://blog.aspect.dev/bzlmod

It'll probably take a bit before things switch over, but I thought I should give a heads, and ask if it'd be something you'd consider supporting with Renovate, especially since I didn't see any other issues mentioning it. As far as I can tell, the need for Renovate with it should be as burning as ever; I haven't seen anything about notifying users about updates on the Bazel side.

Thanks so much for your consideration,
Chris

### If you have any ideas on how this should be implemented, please tell us here.

Package fetching would probably need to be from the [registry](https://github.com/bazelbuild/bazel-central-registry), possibly delegating to Bazel. The actual format is simpler than the current WORKSPACE approach, though it's be in a new file, MODULE.bazel.

### PR DESCRIPTION
**Pull request by cgrindel**
## Changes

- Update the Bazel documentation to include information on Bazel module support.
- Enable the `bazel-module` manager.

## Context

Closes #13658.

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [x] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Review by viceice — COMMENTED** [review]
needs lint fix

**Comment by cgrindel** [comment]
@viceice Thanks. It took me a minute to find the `doc-fix` command.

File: docs/usage/bazel.md
Comment (HonkingGoose) [inline docs/usage/bazel.md:30]:
```suggestion
Here are some important points about Renovate's Bazel registry searches.
Renovate will:

- use _all_ `--registry` flag values found in a workspace's `.bazelrc` file
- use any files that are transitively imported by a `.bazelrc` file
- only use `--registry` flag values that are not associated with [a configuration](https://bazel.build/run/bazelrc#config)
- query the registries in the order that they are found in the `.bazelrc` file
```

File: docs/usage/bazel.md
Comment (HonkingGoose) [inline docs/usage/bazel.md:34]:
```suggestion
In this example, there is a `.bazelrc` file which imports another file called `.registry.bazelrc`.
```

File: docs/usage/bazel.md
Comment (HonkingGoose) [inline docs/usage/bazel.md:103]:
```suggestion
The [`single_version_override`](https://bazel.build/rules/lib/globals/module#single_version_override) is a declaration with many purposes.
```

File: docs/usage/bazel.md
Comment (HonkingGoose) [inline docs/usage/bazel.md:121]:
```suggestion
Any registry values specified in the repository's `.bazelrc` files will be ignored for the `rules_foo` module.
```

File: docs/usage/bazel.md
Comment (HonkingGoose) [inline docs/usage/bazel.md:134]:
```suggestion
If Renovate finds an [`archive_override`](https://bazel.build/rules/lib/globals/module#archive_override) or a [`local_path_override`](https://bazel.build/rules/lib/globals/module#local_path_override), it will ignore the related `bazel_dep`.
```

**Comment by zharinov** [comment]
Seems like we have sort of flip-flop issue with codecov status

File: docs/usage/bazel.md
Comment (HonkingGoose) [inline docs/usage/bazel.md:150]:
```suggestion
Renovate ignores [`multiple_version_override`](https://bazel.build/rules/lib/globals/module#multiple_version_override).
```

Do you think we should explain _why_ Renovate ignores `multiple_version_override`?

File: docs/usage/bazel.md
Comment (HonkingGoose) [inline docs/usage/bazel.md:151]:
```suggestion
`multiple_version_override` does not affect the processing of version updates for a module.
```

File: docs/usage/bazel.md
Comment (HonkingGoose) [inline docs/usage/bazel.md:153]:
```suggestion
## Legacy `WORKSPACE` files
```

File: docs/usage/bazel.md
Comment (HonkingGoose) [inline docs/usage/bazel.md:155]:
```suggestion
Renovate will extract dependencies from:

- `container_pull`
- `oci_pull`
- `git_repository`
- `go_repository`
- `maven_install`
- `http_archive` or `http_file` declarations
```

Let's make a list out of this long sentence.

File: docs/usage/bazel.md
Comment (HonkingGoose) [inline docs/usage/bazel.md:177]: Did you check if monospacing the heading at line 177 and 197 breaks any links? I don't know if we have links in the first place, but you should check.

File: docs/usage/bazel.md
Comment (HonkingGoose) [inline docs/usage/bazel.md:12]: This list should be numbered like this:

```markdown
1. First item
1. Second item
1. Third item
1. Fourth item
```

**Review by HonkingGoose — CHANGES_REQUESTED** [review]
First round of fixes:

- Reduce capitalization in headings
- Use lists
- Rely on the Markdown renderer (or the site build process) to number ordered lists correctly like `1. 2. 3. 4. ....`
- Use simple words and phrases
- Remove duplicate words from list items by creating a platform sentence

File: docs/usage/bazel.md
Comment (cgrindel) [reply docs/usage/bazel.md:22]: The flags can be specified as command line arguments to Bazel or in a bazelrc file. Renovate only cares about the bazelrc files.

File: docs/usage/bazel.md
Comment (cgrindel) [reply docs/usage/bazel.md]: I reworded it a bit.

File: docs/usage/bazel.md
Comment (cgrindel) [reply docs/usage/bazel.md:24]: I don't believe that your suggested edit is quite right. The reader is not checking for the registry entries. Renovate is doing this work.

Comment (cgrindel) [reply docs/usage/bazel.md:24]: I reworded it a bit. Let me know if that works.

File: docs/usage/bazel.md
Comment (cgrindel) [reply docs/usage/bazel.md:151]: Not really. The use case for `multiple_version_override` is very niche. If the reader is using it, they know why they are using it. I believe that the takeaway for the reader is "Renovate won't mess up your configuration".

File: docs/usage/bazel.md
Comment (HonkingGoose) [inline docs/usage/bazel.md:22]:
```suggestion
You can customize the registries your Bazel workspace uses by including [`--registry` CLI flag](https://bazel.build/reference/command-line-reference#flag--registry) entries in your [`.bazelrc` files](https://bazel.build/run/bazelrc).
```

File: docs/usage/bazel.md
Comment (HonkingGoose) [inline docs/usage/bazel.md:23]:
```suggestion
Renovate checks the workspace's `.bazelrc` files for custom registry entries.
```

File: docs/usage/bazel.md
Comment (HonkingGoose) [inline docs/usage/bazel.md:56]:
```suggestion
1. `<https://example.com/custom_registry>`
1. `<https://raw.githubusercontent.com/bazelbuild/bazel-central-registry/main>`
```

File: docs/usage/bazel.md
Comment (HonkingGoose) [inline docs/usage/bazel.md:79]:
```suggestion
Renovate will update the `version` value for a [`bazel_dep`](https://bazel.build/rules/lib/globals/module#bazel_dep) declaration.
```

File: docs/usage/bazel.md
Comment (HonkingGoose) [inline docs/usage/bazel.md:90]:
```suggestion
If Renovate finds a [`git_override`](https://bazel.build/rules/lib/globals/module#git_override), it will ignore the related `bazel_dep` and evaluate the `commit` value at the specified `remote`.
```

File: docs/usage/bazel.md
Comment (HonkingGoose) [inline docs/usage/bazel.md:102]:
```suggestion
If the primary branch has a newer commit than in the list, Renovate will update the `commit` value.
```

Could we go for something like this, where we use simpler words and explain what tool does what?

File: docs/usage/bazel.md
Comment (HonkingGoose) [inline docs/usage/bazel.md:169]: Do we mean Renovate needs all _three_ items, or just _any one_ of them?

File: docs/usage/bazel.md
Comment (HonkingGoose) [inline docs/usage/bazel.md:189]: Same _one_ or _all three_ question applies here.

**Review by HonkingGoose — CHANGES_REQUESTED** [review]
Second round of review. 🙂

File: docs/usage/bazel.md
Comment (cgrindel) [reply docs/usage/bazel.md:169]: The declaration needs all three of the listed items.

File: docs/usage/bazel.md
Comment (cgrindel) [reply docs/usage/bazel.md:189]: All three

**Review by rarkins — DISMISSED** [review]

**Review by viceice — COMMENTED** [review]
waiting for @HonkingGoose approval

File: docs/usage/bazel.md
Comment (HonkingGoose) [inline docs/usage/bazel.md:86]:
```suggestion
In the example above, Renovate evaluates the `0.15.0` version against the repository's registries.
If Renovate finds a newer version, it will update `0.15.0` to match that version.
```

How about something like this?

File: docs/usage/bazel.md
Comment (HonkingGoose) [inline docs/usage/bazel.md:90]:
```suggestion
If Renovate finds a [`git_override`](https://bazel.build/rules/lib/globals/module#git_override), it ignores the related `bazel_dep` entry and instead evaluates the `commit` value at the specified `remote`.
```

File: docs/usage/bazel.md
Comment (HonkingGoose) [inline docs/usage/bazel.md:102]:
```suggestion
If the primary branch has a newer commit than in the list, Renovate updates the `commit` value.
```

File: docs/usage/bazel.md
Comment (HonkingGoose) [inline docs/usage/bazel.md:107]:
```suggestion
The [`single_version_override`](https://bazel.build/rules/lib/globals/module#single_version_override) is a declaration with many purposes.
Renovate only evaluates _two_ attributes from this declaration: `version` and `registry`.
```

The text "is a declaration with many purposes." implies that there are more attributes/purposes, but Renovate can only check some of those attributes, or be used for some of those purposes?

Maybe emphasizing _only_ and _some_ makes it clearer that Renovate won't use the full range of options here?

**Review by HonkingGoose — CHANGES_REQUESTED** [review]
Last round. 😉

File: docs/usage/bazel.md
Comment (cgrindel) [reply docs/usage/bazel.md:107]: That sounds reasonable. In reality, this distinction is true for most of the declarations discussed in this doc. There are attributes on most of these declarations that are not relevant for Renovate.

File: docs/usage/bazel.md
Comment (cgrindel) [reply docs/usage/bazel.md:25]: Good catch.

File: docs/usage/bazel.md
Comment (cgrindel) [reply docs/usage/bazel.md]: I removed the reference to CLI flag. If a Bazel workspace maintainer uses custom registry settings, they will use `.bazelrc` files to specify these values. I don't think Introducing the CLI flag concept helps readers understand how Renovate will use the values.

File: docs/usage/bazel.md
Comment (HonkingGoose) [inline docs/usage/bazel.md:32]: We still mention the word `flag` here, is that intended?

File: docs/usage/bazel.md
Comment (HonkingGoose) [inline docs/usage/bazel.md:108]: This is a hard sentence to read, and also for me to rewrite. We're using the word `specified` three times.

Open for suggestions. 😉

**Review by HonkingGoose — CHANGES_REQUESTED** [review]
This looks way better after your voice passes. 🥳 There are still some small things to fix though.

File: docs/usage/bazel.md
Comment (cgrindel) [reply docs/usage/bazel.md:32]: No. Good catch.

File: docs/usage/bazel.md
Comment (cgrindel) [reply docs/usage/bazel.md]: I trimmed it down. Does that work?

File: docs/usage/bazel.md
Comment (HonkingGoose) [inline docs/usage/bazel.md:24]: The current link points to the GitHub repository for the Bazel Central Registry.

I would expect to land on the registry, based on the link name.

Options:

1. Keep URL to GitHub repository and edit the link name to match the destination
2. Change URL to land on actual registry

```suggestion
If no registries are found, Renovate defaults to the [Bazel Central Registry](https://registry.bazel.build/).
```

I'll let you decide, as you know more about the Bazel users. 😉

**Review by HonkingGoose — APPROVED** [review]
Ready to merge after fixing this thing. 😉

File: docs/usage/bazel.md
Comment (cgrindel) [reply docs/usage/bazel.md]: The actual registry is the GitHub repo. The URL that you suggested is for the catalog. It is a little confusing. I updated the URL to be the one that they advertise in the doc. It is a redirect to the URL that I had.

**Review by HonkingGoose — APPROVED** [review]

**Review by rarkins — APPROVED** [review]

**Pull request merged**
